### DISCUSSION TITLE
Questions about paper 'grand-potential-based phase-field approach'

### DISCUSSION THREAD

**Discussion by GuoChi-Li** [idaholab/moose#22014] [Q&A Modules: Phase field] [answered]
Hi MOOSE developers,
        I have questions about the paper 'Solidification and grain formation in alloys: a 2D application of the grand-potential-based phase-field approach' and I have sent an email to the author of this paper. However, I didn't get a reply, so I try to ask questions here. Any help is appreciated. The following texts are the email context. 
       I am a Ph.D. student majoring in materials science and engineering and I am new to this field. Recently, I read your work published in Modelling and Simulation in Materials Science and Engineering journal. The title of this paper is ‘Solidification and grain formation in alloys: a 2D application of the grand-potential-based phase-field approach.’
I think this is an excellent starting point for my research. I want to simulate the dendrite growth under different laser heating paths. Therefore, I use the demo file in the MOOSE directory (moose/modules/phase_field/examples/anisotropic_interfaces/GrandPotentialPlanarGrowth.i) as my start and try to reproduce your work, especially figure 12, figure 13, and figure 14 in the paper. 
After I modify the input file, try to implement the antitrapping current and Langevin noise in the kernel. I find it hard to converge, so I change the input parameters. I am using the parameters in ‘GrandPotentialPlanarGrowth.i’ and only add antitrapping current and Langevin noise. The attachments are my input file and the final output log. Also, I have several questions related to this work.
1.	The mu in the code (in GenericConstantMaterial) is the bulk energy coefficient, right?
2.	How do you determine the ‘-aW’ in the antitrapping current term? I see this value is 0.8 in ‘GrandPotentialAnisotropyAntitrap.i’ but I do see you use different values (aW = 0.113, 0.095, 0.071) when you discuss the effect of antitrapping current in figure 10. I am not sure which value I should use.
3.	How do you implement the wavy surface in figure 12? Is it possible to use 'FunctionIC' to implement this in MOOSE?
4.	Is there any possible ways to speed up the convergence?

Thank you for your time. Any help is appreciated.
Sincerely,

Guo-Chi Li

[Input_file.log](https://github.com/idaholab/moose/files/9497168/Input_file.log)
[output_log.log](https://github.com/idaholab/moose/files/9497170/output_log.log)

**Comment by SudiptaBiswas** [comment, marked as answer]
> The mu in the code (in GenericConstantMaterial) is the bulk energy coefficient, right?

That is correct. 

> How do you determine the ‘-aW’ in the antitrapping current term? I see this value is 0.8 in ‘GrandPotentialAnisotropyAntitrap.i’ but I do see you use different values (aW = 0.113, 0.095, 0.071) when you discuss the effect of antitrapping current in figure 10. I am not sure which value I should use.

`W` is the interface thickness, `a` is the antitrapping coefficients. The paper presents a parametric study on how this affects solidification behavior. The actual values could vary with the problem setup. The paper discusses this briefly (check out Appendix C).  

> How do you implement the wavy surface in figure 12? Is it possible to use 'FunctionIC' to implement this in MOOSE?

You probably meant figure 14. I used a Sine function to set it up. 

> Is there any possible ways to speed up the convergence? 

In general, complex alloy solidification problems can be hard to solve. You can non-dimensionalize your model parameters to improve that.

**Reply by GuoChi-Li** [reply]
Hi SudiptaBiswas,

Thanks for your detailed explanations. I tried to reply to you in an organized way.

> 'W' is the interface thickness, 'a' is the antitrapping coefficients. The paper presents a parametric study on how this affects solidification behavior. The actual values could vary with the problem setup. The paper discusses this briefly (check out Appendix C).

Yes. I have read that part. I need to find those references since I am unfamiliar with this part. I will check them out.

> You probably meant figure 14. I used a Sine function to set it up.

Yes. I am discussing figure 14. I mean, I think you probably implement Sine function using something like parsed_function or FunctionIC in MOOSE. Is that right?

> In general, complex alloy solidification problems can be hard to solve. You can non-dimensionalize your model parameters to improve that.

Thanks. I will look into this.

Is it possible to provide the input files you used in this work? It is ok if you still need them for your future publications and can't release them now.

Sincerely,

Guo-Chi Li

**Reply by SudiptaBiswas** [reply]
Yeah, you can use `ParsedFunction` and `FunctionIC` to set up the initial condition. The input files checked in as examples are a good place to start. Try to modify one thing at a time as you progress towards a different solidification scenario.

**Reply by GuoChi-Li** [reply]
Thank you for your help. I will try to do it.
